decimal64.go

```diff
@@ -457,7 +457,13 @@ func numDecimalDigits(n uint64) int {
 
 // propagateNan returns true if flavor1 should be propagated else false if flavor2 should be propagated
 func propagateNan(dp, ep *decParts) *Decimal64 {
-	if dp.fl == flSNaN || dp.fl == flQNaN {
+	if dp.fl == flSNaN {
+		return dp.dec
+	}
+	if ep.fl == flSNaN {
+		return ep.dec
+	}
+	if dp.fl == flQNaN {
```

Comment:
is there any reason this statement can't be grouped with the first condition?

ie.

```go
if dp.fl == flSNaN || dp.fl == flQNaN {
  return dp.dec
}
```

(I've only been peripherally watching the past PR's so apologies if there are stylistic reasons of which I'm unaware)

Reply:
I'm guessing the second test must precede the third.

Reply:
The first SNaN that's enountered must be returned first, then if there are no SNans the first QNan must be returned,
Grouping doesn't work, neither do switch statements
order of 'importance':
`dp SNaN > ep SNaN > dp QNaN> ep QNaN`

Reply:
ah - of course, apologies!